packages/valory/skills/hello_world_abci/rounds.py

```diff
@@ -95,6 +96,30 @@ def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
         return None
 
 
+class CollectRandomnessRound(
+    CollectSameUntilThresholdRound, HelloWorldABCIAbstractRound
+):
+    """A round for generating randomness"""
+
+    round_id = "collect_randomness"
+    allowed_tx_type = CollectRandomnessPayload.transaction_type
+    payload_attribute = "randomness"
+
+    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
+        """Process the end of the block."""
+        if self.threshold_reached:
+            synchronized_data = self.synchronized_data.update(
+                participant_to_randomness=self.collection,
+                most_voted_randomness=self.most_voted_payload,
```

Comment:
Just use collection and selection keys.

Reply:
Will action in an upcoming 'improvement' PR once merged.